modules/fedimint-mint/src/lib.rs

```diff
@@ -145,17 +145,38 @@ pub struct VerifiedNotes {
 }
 
 #[derive(Debug)]
-pub struct MintGen;
+pub struct MintCommonGen;
 
-#[apply(async_trait_maybe_send!)]
-impl ModuleGen for MintGen {
+impl CommonModuleGen for MintCommonGen {
     const KIND: ModuleKind = KIND;
-    const DATABASE_VERSION: DatabaseVersion = DatabaseVersion(0);
 
-    fn decoder(&self) -> Decoder {
+    fn decoder() -> Decoder {
         <Mint as ServerModule>::decoder()
     }
 
+    fn hash_client_module(
+        config: serde_json::Value,
+    ) -> anyhow::Result<bitcoin_hashes::sha256::Hash> {
+        serde_json::from_value::<MintClientConfig>(config)?.consensus_hash()
+    }
+}
+
+#[derive(Debug)]
+pub struct MintClientGen;
+
+#[apply(async_trait_maybe_send!)]
+impl ClientModuleGen for MintClientGen {
+    type Common = MintCommonGen;
+}
+
+#[derive(Debug)]
+pub struct MintGen;
```

Comment:
Nit: `MintServerGen` is probably better here

Reply:
I'll rename in another PR to avoid too many merge conflicts.